lib/spack/spack/cmd/develop.py

```diff
@@ -84,86 +79,152 @@ def _retrieve_develop_source(spec: spack.spec.Spec, abspath: str) -> None:
     package.stage.steal_source(abspath)
 
 
-def develop(parser, args):
-    # Note: we could put develop specs in any scope, but I assume
-    # users would only ever want to do this for either (a) an active
-    # env or (b) a specified config file (e.g. that is included by
-    # an environment)
-    # TODO: when https://github.com/spack/spack/pull/35307 is merged,
-    # an active env is not required if a scope is specified
-    env = spack.cmd.require_active_env(cmd_name="develop")
-    if not args.spec:
-        if args.clone is False:
-            raise SpackError("No spec provided to spack develop command")
-
-        # download all dev specs
-        for name, entry in env.dev_specs.items():
-            path = entry.get("path", name)
-            abspath = spack.util.path.canonicalize_path(path, default_wd=env.path)
-
-            if os.path.exists(abspath):
-                msg = "Skipping developer download of %s" % entry["spec"]
-                msg += " because its path already exists."
-                tty.msg(msg)
-                continue
+def assure_concrete_spec(env: spack.environment.Environment, spec: spack.spec.Spec):
+    version = spec.versions.concrete_range_as_version
+    if not version:
+        # first check environment for a matching concrete spec
+        matching_specs = env.all_matching_specs(spec)
+        if matching_specs:
+            version = matching_specs[0].version
+            test_spec = spack.spec.Spec(f"{spec}@{version}")
+            for m_spec in matching_specs:
+                if not test_spec.satisfies(m_spec):
+                    raise SpackError(
+                        f"{spec.name}: has conflicting specs and only one develop spec per package "
+                        "can be defined. Please unify the versions or reduce the specs to use "
+                        "`spack develop`"
```

Comment:
This last sentence doesn't scan easily for me -- is the second suggestion supposed to be "or do not `spack develop` {spec.name}"?

Reply:
Well no, but that is fine too. It's supposed to read, make the versions the same or delete specs until you only have the same version to use spack develop. I suppose that second one is redundant.

Reply:
@becker33  I've updated the error message:
``` console
$ spack develop zlib-ng
==> Error: zlib-ng: has multiple concrete instances in the graph that can't be satisified by a single develop spec. To use `spack develop` ensure one of the following:
 a) zlib-ng nodes can satisfy the same develop spec (minimally this means they all share the same version)
 b) Provide a concrete develop spec (zlib-ng@[version]) to clearly indicate what should be developed
```